### PR TITLE
Droth 363 saving of light history point assets

### PR DESCRIPTION
**Pull request by oskarmalmstrom**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetOperations.scala [hunk 1]
```diff
@@ -237,19 +237,27 @@ trait PointAssetOperations {
...
 
   def expire(id: Long, username: String): Long = {
     withDynSession {
-      Queries.updateAssetModified(id, username).first
-      sqlu"update asset set valid_to = sysdate where id = $id".first
+      expireWihoutTransaction(id, username)
     }
   }
 
+  def expireWihoutTransaction(id: Long, username: String) = {
```
Comment (oskarmalmstrom) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetOperations.scala]: typo in without

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala [hunk 2]
```diff
@@ -54,9 +65,9 @@ class ObstacleService(val roadLinkService: RoadLinkService) extends PointAssetOp
     createPersistedAsset(asset, adjustment)
   }
 
-  private def adjustmentOperation(persistedAsset: PersistedAsset, adjustment: AssetAdjustment): Long = {
+  private def adjustmentOperation(persistedAsset: PersistedAsset, adjustment: AssetAdjustment, roadLink: RoadLink): Long = {
     val updated = IncomingObstacle(adjustment.lon, adjustment.lat, adjustment.linkId, persistedAsset.obstacleType)
-    OracleObstacleDao.update(adjustment.assetId, updated, adjustment.mValue, "vvh_generated", persistedAsset.municipalityCode, Some(adjustment.vvhTimeStamp), persistedAsset.linkSource)
+    updateWithoutTransaction(adjustment.assetId, updated, roadLink.geometry, persistedAsset.municipalityCode, "vvh_generated", persistedAsset.linkSource)
   }
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala]: Maybe "Some(adjustment.vvhTimeStamp)" shouldn't be excluded  from update (this is related with changeInfo.vvhTimeStamp)

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/pointasset/oracle/OraclePedestrianCrossingDao.scala [hunk 3]
```diff
@@ -53,16 +53,38 @@ object OraclePedestrianCrossingDao {
...
+  def create(crossing: IncomingPedestrianCrossing, mValue: Double, username: String, municipality: Long, adjustedTimestamp: Long, linkSource: LinkGeomSource,
+             createdByFromUpdate: Option[String] = Some(""),
+             createdDateTimeFromUpdate: Option[DateTime]): Long = {
+    val id = Sequences.nextPrimaryKeySeqValue
+    val lrmPositionId = Sequences.nextLrmPositionPrimaryKeySeqValue
+    sqlu"""
+      insert all
+        into asset(id, asset_type_id, created_by, created_date, municipality_code, modified_by, modified_date)
+        values ($id, 200, $createdByFromUpdate, $createdDateTimeFromUpdate, ${municipality}, $username, sysdate)
+
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/pointasset/oracle/OraclePedestrianCrossingDao.scala]: Maybe we could create a global asset_type_id = 200

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/DirectionalTrafficSignService.scala [hunk 4]
```diff
@@ -34,18 +34,30 @@ class DirectionalTrafficSignService(val roadLinkService: RoadLinkService) extend
...
+
+  def updateWithoutTransaction(id: Long, updatedAsset: IncomingDirectionalTrafficSign, geometry: Seq[Point], municipality: Int, username: String, linkSource: LinkGeomSource): Long = {
+    val oldAsset = getPersistedAssetsByIdsWithoutTransaction(Set(id)).headOption
+    val mValue = GeometryUtils.calculateLinearReferenceFromPoint(Point(updatedAsset.lon, updatedAsset.lat), geometry)
+    oldAsset match {
+      case Some(old) if old.bearing != updatedAsset.bearing || ( old.lat != updatedAsset.lat || old.lon != updatedAsset.lon) =>
+        expireWihoutTransaction(id, username)
+        OracleDirectionalTrafficSignDao.create(setAssetPosition(updatedAsset, geometry, mValue), mValue, municipality, username,old.createdBy, old.createdAt)
+      case _ =>
+        OracleDirectionalTrafficSignDao.update(id, setAssetPosition(updatedAsset, geometry, mValue), mValue, municipality, username)
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/DirectionalTrafficSignService.scala]: Maybe this "case _"  should be replace by  "case Some(old)"

**Comment by oskarmalmstrom** [comment]
ok

**Comment by joanacoelho** [comment]
Looks good

**Pull request merged**